core/controllers/editor.py

```diff
@@ -272,8 +273,8 @@ class ExplorationHandler(EditorHandler):
     def _get_exploration_data(self, exploration_id, version=None):
         """Returns a description of the given exploration."""
         try:
-            exploration = exp_services.get_exploration_by_id(
-                exploration_id, version=version)
+            exploration = exp_services.get_exp_with_draft_applied(
```

Comment:
How can you safely apply the draft changes without first checking if the draft version is valid?